core/arch/x86/decode.c

```diff
@@ -2298,6 +2487,18 @@ decode_common(dcontext_t *dcontext, byte *pc, byte *orig_pc, instr_t *instr)
     di.len = (int)(next_pc - pc);
     di.opcode = info->type; /* used for opnd_create_immed_float_for_opcode */
 
+    /* The upper DR_TUPLE_TYPE_BITS bits of the flags field are the evex tuple type. */
+    di.tuple_type = (dr_tuple_type_t)(info->flags >> DR_TUPLE_TYPE_BITPOS);
```

Comment:
So this assumes it's the top bits rather than using DR_TYPE_TYPE_MASK.  Why have the mask defined if it's not used?  Worth a comment (or, use the mask, or remove the mask and comment about the assumption).

Reply:
Removed the mask.